server/com.dexels.navajo.compiler.tsl/src/com/dexels/navajo/compiler/tsl/internal/BundleQueueComponent.java

```diff
@@ -64,25 +65,39 @@ public void clearDependencyAnalyzer(DependencyAnalyzer depa) {
     @Override
     public void enqueueScript(final String script, final String path) {
         executor.execute(() -> {
-		    List<String> failures = new ArrayList<>();
-		    List<String> success = new ArrayList<>();
-		    List<String> skipped = new ArrayList<>();
-		    logger.info("Eagerly compiling: {}", script);
-		    try {
-		        bundleCreator.createBundle(script, failures, success, skipped, true, keepIntermediateFiles);
-		        bundleCreator.installBundle(script, failures, success, skipped, true);
-		        if (!skipped.isEmpty()) {
-		            logger.info("Script compilation skipped: {}", script);
-		        }
-		        if (!failures.isEmpty()) {
-		            logger.info("Script compilation failed: {}", script);
-		        }
-
-		    } catch (Throwable e) {
-		        logger.error("Error: ", e);
-		    }
+		    compileScript(script, path);
 		});
     }
+    
+    /* (non-Javadoc)
+     * @see com.dexels.navajo.compiler.tsl.internal.BundleQueue#compileScript(java .lang.String) */
+    @Override
+    public synchronized boolean compileScript(final String script, final String path) {
+        boolean compilationSuccess = true;
+        List<String> failures = new ArrayList<>();
+        List<String> success = new ArrayList<>();
+        List<String> skipped = new ArrayList<>();
+        logger.info("Eagerly compiling: {}", script);
+        try {
+            bundleCreator.createBundle(script, failures, success, skipped, true, keepIntermediateFiles);
+            bundleCreator.installBundle(script, failures, success, skipped, true);
+            if (!skipped.isEmpty()) {
+                compilationSuccess = false;
+                logger.info("Script compilation skipped: {}", script);
+            }
+            if (!failures.isEmpty()) {
+                compilationSuccess = false;
+                logger.info("Script compilation failed: {}", script);
+            }
+            ensureScriptDependencies(script);
+            enqueueDependentScripts(script);
```

Comment:
Why do you deal with dependents/dependencies when there is a compilation failure?

Reply:
changed to only to this upon compilationSuccess